### ISSUE TITLE
Improve CI robustness

### ISSUE THREAD

**Issue by ajm-ska** [CARTAvis/carta-frontend#2142]
**Describe the bug**
Our carta-frontend CI occasionally fails on the "Production build stage". The error being that GSL header files are missing:

> gsl_wrapper.cc:2:10: fatal error: 'gsl/gsl_math.h' file not found
> #include "gsl/gsl_math.h"
>          ^~~~~~~~~~~~~~~~
> 1 error generated.

The issue actually originates during the previous "Build libs" stage, specifically in the `build_gsl.sh` script. But that stage is issued as a success even though the GSL libraries failed to build.

The problem is that our self-hosted Github Runners fail to download the GSL source code from http://ftpmirror.gnu.org/gsl/gsl-2.6.tar.gz

The download problem only happens with GSL and not the other library builds (ast, zfp, zstd), so it suggests it is not a network problem at our end. It typically occurs during a PR. Before #2139 the CI performed builds on 3 node versions in parallel (Node v12, v14, and v16). However, when there is a PR, there are actually two sets of builds; one with the current branch, and one of what it would be if merged into the dev branch. Therefore would be 6 parallel builds. Not to mention the possibility of other builds from other branches and PRs. Therefore, I suspect there could be a lot of requests during a short period of time to the GSL download server which could be considered suspicious and access from our IP address is likely blocked by them for a short amount of time, perhaps through a service such as fail2ban.

**To Reproduce**
1. Make several PRs and commits.
2. Wait for the CI to fail on the Production build stage.
3. See error

**Expected behavior**
The CI should not fail due to a missing "gsl/gsl_math.h", or if it fails due to that, the CI should have reported the issue and already exited at that stage.

I will improve the `build_gsl.sh` script to address these issues.
